src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -988,22 +987,29 @@ private void WriteString(in DbRow row, Utf8JsonWriter writer)
                     arrayItemsCount = row.SizeOrLength;
                     numberOfRowsForValues += row.NumberOfRows;
                 }
-                else if (tokenType == JsonTokenType.PropertyName)
+                else if (IsStringLike(tokenType)) // Equivalent to (tokenType == JsonTokenType.PropertyName || tokenType == JsonTokenType.String)
```

Comment:
Why collapse this?  The code structure was "JSON structure" (start array, start object, end array, end object, property name) then "JSON value" (everything else).

Reply:
Primarily to avoid duplication between the property name and string case. I can keep it separate if you prefer. Also, a check like this is faster for the more common property name/string cases.

Reply:
> I can keep it separate if you prefer. 

Yes, please :smile: